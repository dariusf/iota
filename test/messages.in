Lobby setSlot("Person", Object clone)
Person hello := 1
Person talk := method(name, writeln(name charAt(2)))
Person hello
if (true, Person talk("billy"), Person talk("mandy"))
writeln(Person slotNames)