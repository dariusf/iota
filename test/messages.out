Lobby.send('Lobby').send('setSlot', (IoStringWrapper('Person')), (Lobby.send('Object').send('clone')));
Lobby.send('Person').send('setSlot', (IoStringWrapper('hello')), (IoNumberWrapper(1)));
Lobby.send('Person').send('setSlot', (IoStringWrapper('talk')), (Lobby.send('method', 'name', IoThunk(function (locals) {
    return locals.send('writeln', (locals.send('name').send('charAt', (IoNumberWrapper(2)))));
}))));
Lobby.send('Person').send('hello');
Lobby.send('if', (Lobby.send('true')), IoThunk(function () {
    return Lobby.send('Person').send('talk', (IoStringWrapper('billy')));
}), IoThunk(function () {
    return Lobby.send('Person').send('talk', (IoStringWrapper('mandy')));
}));
Lobby.send('writeln', (Lobby.send('Person').send('slotNames')));